packages/sdk38/src/lcdclient.ts

```diff
@@ -0,0 +1,350 @@
+    setupModuleC?: LcdModuleSetup<C>,
+    setupModuleD?: LcdModuleSetup<D>,
+    setupModuleE?: LcdModuleSetup<E>,
+    setupModuleF?: LcdModuleSetup<F>,
+    setupModuleG?: LcdModuleSetup<G>,
+    setupModuleH?: LcdModuleSetup<H>,
+  ): any {
+    const client = new LcdClient(options.apiUrl, options.broadcastMode);
+
+    const modules = new Array<LcdModule>();
+    if (setupModuleA) modules.push(setupModuleA(client));
+    if (setupModuleB) modules.push(setupModuleB(client));
+    if (setupModuleC) modules.push(setupModuleC(client));
+    if (setupModuleD) modules.push(setupModuleD(client));
+    if (setupModuleE) modules.push(setupModuleE(client));
+    if (setupModuleF) modules.push(setupModuleF(client));
+    if (setupModuleG) modules.push(setupModuleG(client));
+    if (setupModuleH) modules.push(setupModuleH(client));
+    for (const module of modules) {
+      assert(isNonNullObject(module), `Module must be a non-null object`);
+      for (const key in module) {
+        assert(typeof key == "string", `Found non-string module key: ${key}`);
+        (client as any)[key] = module[key];
+      }
+    }
+
+    return client;
+  }
+
+  private readonly client: AxiosInstance;
+  private readonly broadcastMode: BroadcastMode;
+
+  /**
+   * Creates a new client to interact with a Cosmos SDK light client daemon.
+   * This class tries to be a direct mapping onto the API. Some basic decoding and normalizatin is done
+   * but things like caching are done at a higher level.
+   *
+   * When building apps, you should not need to use this class directly. If you do, this indicates a missing feature
+   * in higher level components. Feel free to raise an issue in this case.
+   *
+   * @param apiUrl The URL of a Cosmos SDK light client daemon API (sometimes called REST server or REST API)
+   * @param broadcastMode Defines at which point of the transaction processing the postTx method (i.e. transaction broadcasting) returns
+   */
+  public constructor(apiUrl: string, broadcastMode = BroadcastMode.Block) {
+    const headers = {
+      post: { "Content-Type": "application/json" },
+    };
+    this.client = axios.create({
+      baseURL: apiUrl,
+      headers: headers,
+    });
+    this.broadcastMode = broadcastMode;
+  }
+
+  public async get(path: string): Promise<any> {
+    const { data } = await this.client.get(path).catch(parseAxiosError);
+    if (data === null) {
+      throw new Error("Received null response from server");
+    }
+    return data;
+  }
+
+  public async post(path: string, params: any): Promise<any> {
+    if (!isNonNullObject(params)) throw new Error("Got unexpected type of params. Expected object.");
+    const { data } = await this.client.post(path, params).catch(parseAxiosError);
+    if (data === null) {
+      throw new Error("Received null response from server");
+    }
+    return data;
+  }
+
+  // The /auth endpoints
+
+  public async authAccounts(address: string): Promise<AuthAccountsResponse> {
```

Comment:
This is actually a module. Sure, everyone needs it, but interesting to see if this could be unified with the other modules.

Reply:
Yeah good point. It better shows how things are supposed to work

Reply:
Implemented in packages/sdk38/src/lcdapi/auth.ts. The basic CosmosClient now uses

```ts
  protected readonly lcdClient: LcdClient & AuthExtension;
```

and


```ts
    this.lcdClient = LcdClient.withExtensions(
      { apiUrl: apiUrl, broadcastMode: broadcastMode },
      setupAuthExtension,
    );
```